### PR TITLE
[path_provider] add getApplicationLibraryDirectory

### PR DESCRIPTION
**Pull request by jerryzhoujw**
## Description

For iOS, `NSLibraryDirectory` are more used for application data like sqlite.db or images which would like to be capacity to persistent, backed up and not visible in iTunes File Sharing when `UIFileSharingEnabled` is `TRUE` in app's plist file.

From StackOverFlow discuss: [Path Directory Usable in iOS](https://stackoverflow.com/questions/7268299/path-directory-usable-in-ios)

- `NSDocumentDirectory` is `Documents/`     (persistent, backed up, may be visible in iTunes)
- `NSLibraryDirectory` is `Library/`     (persistent, backed up, not visibe to the user)
- `NSCachesDirectory` is `Library/Caches/`     (not backed up, may be cleared by system)

From iOS framework:

- `NSLibraryDirectory`,   // various documentation, support, and configuration files, resources (Library)
- `NSApplicationSupportDirectory` = 14, // location of application support files (plug-ins, etc) (Library/Application Support)

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/path_provider/lib/path_provider.dart [hunk 1]
```diff
@@ -50,6 +50,34 @@ Future<Directory> getApplicationSupportDirectory() async {
   return Directory(path);
 }
 
+/// Path to the dictory where applcation can
```
Comment (collinjackson) [inline packages/path_provider/lib/path_provider.dart]:
```suggestion
/// Path to the directory where application can store
```

File: packages/path_provider/lib/path_provider.dart [hunk 2]
```diff
@@ -50,6 +50,34 @@ Future<Directory> getApplicationSupportDirectory() async {
...
+/// `persistent, backed up, not visibe to the user`, like application sqlite.db
```
Comment (collinjackson) [inline packages/path_provider/lib/path_provider.dart]:
```suggestion
/// `persistent, backed up, not visible to the user` files, such as sqlite.db.
```

**Comment by jerryzhoujw** [comment]
> Should we call this `getLibraryDirectory` to match the naming on iOS?

yes we can

**Comment by jerryzhoujw** [comment]
@collinjackson Do I better to not do `git rebase`  after PR create ? I have asked [someone](https://stackoverflow.com/questions/57506240/should-i-git-rebase-after-fix-pr-issue/57506336?noredirect=1#comment101481819_57506336), but maybe I do better ask review directly

**Comment by ened** [comment]
@jerryzhoujw Great PR. 

@collinjackson can we get this reviewed again soon?

**Pull request merged**